### PR TITLE
DUPLO-6168, DUPLO-13189 fix: update ECS task definition example with AWS supported port mapping field

### PR DESCRIPTION
**Pull request by Matheusbafutto**
## Overview

The correct field for defining container port mappings is `PortMappings` instead of `ContainerMappings`.

## Summary of changes

This PR does the following:

- Adjust ECS task definition example
- Update documentation

## Testing performed

- [ ] Using unit tests
- [x] Manually, on my local system
- [ ] Manually, on a remote test system

## Describe any breaking changes

No

## Sanity Checking

Generated TF state:
```json
   {
      "mode": "managed",
      "type": "duplocloud_ecs_task_definition",
      "name": "my_ecs_service2",
      "provider": "provider[\"registry.terraform.io/duplocloud/duplocloud\"]",
      "instances": [
        {
          "schema_version": 0,
          "attributes": {
            "container_definitions": "[{\"CredentialSpecs\":[],\"Environment\":[{\"Name\":\"NGINX_HOST\",\"Value\":\"foo\"}],\"Essential\":true,\"Image\":\"nginx:latest\",\"LogConfiguration\":{\"LogDriver\":{\"Value\":\"awslogs\"},\"Options\":{\"awslogs-create-group\":\"true\",\"awslogs-group\":\"/ecs/duploservices-non-default-tfmyservice\",\"awslogs-region\":\"us-west-2\",\"awslogs-stream-prefix\":\"ecs\"},\"SecretOptions\":[]},\"Name\":\"default1\",\"PortMappings\":[{\"ContainerPort\":80,\"HostPort\":80,\"Protocol\":{\"Value\":\"tcp\"}}]}]",
            ...
          },
          ...
        }
      ]
    }
```

Formatted container definitions
```json
{
    "Environment": [
        {
            "Name": "NGINX_HOST",
            "Value": "foo"
        }
    ],
   ...
    "PortMappings": [
        {
            "ContainerPort": 80,
            "HostPort": 80,
            "Protocol": {
                "Value": "tcp"
            }
        }
    ]
}
```

___

## **Type**
enhancement, documentation

___

## **Description**
- Corrected the use of `PortMappings` in ECS task definition examples and documentation.
- Updated CHANGELOG, documentation, and Terraform examples to reflect the correct usage of `PortMappings` instead of `ContainerMappings`.
- Fixed typos and incorrect field names in ECS task definition properties.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Documentation
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>CHANGELOG.md
</strong><dd><code>Update CHANGELOG for ECS Task Definition Corrections</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
CHANGELOG.md

<li>Updated <code>duplocloud_ecs_task_definition</code> examples and documentation in <br>the CHANGELOG.<br> <li> Fixed property typos related to <code>PortMappings</code> in <code>container_definitions</code>.<br> <br>

</details>

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/397/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4ed"></a></td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ecs_task_definition.md
</strong><dd><code>Documentation Fix for ECS Task Definition</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
docs/resources/ecs_task_definition.md

<li>Corrected the <code>PortMappings</code> field in the ECS task definition <br>documentation.<br> <li> Replaced incorrect <code>ContainerMappings</code> with <code>PortMappings</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/397/files#diff-6555f19f2f8195e7a1475c5ca7a1df5a15aa9479bbbc24bf67713b408b234f73"></a></td>

</tr>                    
</table></td></tr><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>resource.tf
</strong><dd><code>Correct ECS Task Definition Example with `PortMappings`</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
examples/resources/duplocloud_ecs_task_definition/resource.tf

<li>Updated ECS task definition example to use <code>PortMappings</code>.<br> <li> Corrected field names from <code>ContainerPorts</code> and <code>HostPort</code> to <br><code>ContainerPort</code> and <code>HostPort</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/397/files#diff-fcfb692bee3d99d8581f62a210f15d9057d41bb5e93e1959dcee2bf6bd4e7e73"></a></td>

</tr>                    
</table></td></tr></tr></tbody></table><hr>

<details> <summary><strong>✨ Usage guide:</strong></summary><hr> 

**Overview:**
The `describe` tool scans the PR code changes, and generates a description for the PR - title, type, summary, walkthrough and labels. The tool can be triggered [automatically](https://github.com/Codium-ai/pr-agent/blob/main/Usage.md#github-app-automatic-tools) every time a new PR is opened, or can be invoked manually by commenting on a PR.

When commenting, to edit [configurations](https://github.com/Codium-ai/pr-agent/blob/main/pr_agent/settings/configuration.toml#L46) related to the describe tool (`pr_description` section), use the following template:
```
/describe --pr_description.some_config1=... --pr_description.some_config2=...
```
With a [configuration file](https://github.com/Codium-ai/pr-agent/blob/main/Usage.md#working-with-github-app), use the following template:
```
[pr_description]
some_config1=...
some_config2=...
```

<table><tr><td><details> <summary><strong> Enabling\disabling automation </strong></summary><hr>

- When you first install the app, the [default mode](https://github.com/Codium-ai/pr-agent/blob/main/Usage.md#github-app-automatic-tools) for the describe tool is:
```
pr_commands = ["/describe --pr_description.add_original_user_description=true" 
                         "--pr_description.keep_original_user_title=true", ...]
```
meaning the `describe` tool will run automatically on every PR, will keep the original title, and will add the original user description above the generated description. 

- Markers are an alternative way to control the generated description, to give maximal control to the user. If you set:
```
pr_commands = ["/describe --pr_description.use_description_markers=true", ...]
```
the tool will replace every marker of the form `pr_agent:marker_name` in the PR description with the relevant content, where `marker_name` is one of the following:
  - `type`: the PR type.
  - `summary`: the PR summary.
  - `walkthrough`: the PR walkthrough.

Note that when markers are enabled, if the original PR description does not contain any markers, the tool will not alter the description at all.
        

</details></td></tr>

<tr><td><details> <summary><strong> Custom labels </strong></summary><hr>

The default labels of the `describe` tool are quite generic: [`Bug fix`, `Tests`, `Enhancement`, `Documentation`, `Other`].

If you specify [custom labels](https://github.com/Codium-ai/pr-agent/blob/main/docs/DESCRIBE.md#handle-custom-labels-from-the-repos-labels-page-gem) in the repo's labels page or via configuration file, you can get tailored labels for your use cases.
Examples for custom labels:
- `Main topic:performance` - pr_agent:The main topic of this PR is performance
- `New endpoint` - pr_agent:A new endpoint was added in this PR
- `SQL query` - pr_agent:A new SQL query was added in this PR
- `Dockerfile changes` - pr_agent:The PR contains changes in the Dockerfile
- ...

The list above is eclectic, and aims to give an idea of different possibilities. Define custom labels that are relevant for your repo and use cases.
Note that Labels are not mutually exclusive, so you can add multiple label categories.
Make sure to provide proper title, and a detailed and well-phrased description for each label, so the tool will know when to suggest it.        

</details></td></tr>

<tr><td><details> <summary><strong> Inline File Walkthrough 💎</strong></summary><hr>

For enhanced user experience, the `describe` tool can add file summaries directly to the "Files changed" tab in the PR page.
This will enable you to quickly understand the changes in each file, while reviewing the code changes (diffs).

To enable inline file summary, set `pr_description.inline_file_summary` in the configuration file, possible values are:
- `'table'`: File changes walkthrough table will be displayed on the top of the "Files changed" tab, in addition to the "Conversation" tab.
- `true`: A collapsable file comment with changes title and a changes summary for each file in the PR.
- `false` (default): File changes walkthrough will be added only to the "Conversation" tab.
<tr><td><details> <summary><strong> Utilizing extra instructions</strong></summary><hr>

The `describe` tool can be configured with extra instructions, to guide the model to a feedback tailored to the needs of your project.

Be specific, clear, and concise in the instructions. With extra instructions, you are the prompter. Notice that the general structure of the description is fixed, and cannot be changed. Extra instructions can change the content or style of each sub-section of the PR description.

Examples for extra instructions:
```
[pr_description] 
extra_instructions="""
- The PR title should be in the format: '<PR type>: <title>'
- The title should be short and concise (up to 10 words)
- ...
"""
```
Use triple quotes to write multi-line instructions. Use bullet points to make the instructions more readable.

</details></td></tr>

<tr><td><details> <summary><strong> More PR-Agent commands</strong></summary><hr> 

> To invoke the PR-Agent, add a comment using one of the following commands:  
> - **/review**: Request a review of your Pull Request.   
> - **/describe**: Update the PR title and description based on the contents of the PR.   
> - **/improve [--extended]**: Suggest code improvements. Extended mode provides a higher quality feedback.   
> - **/ask \<QUESTION\>**: Ask a question about the PR.   
> - **/update_changelog**: Update the changelog based on the PR's contents.   
> - **/add_docs** 💎: Generate docstring for new components introduced in the PR.   
> - **/generate_labels** 💎: Generate labels for the PR based on the PR's contents.   
> - **/analyze** 💎: Automatically analyzes the PR, and presents changes walkthrough for each component.   

>See the [tools guide](https://github.com/Codium-ai/pr-agent/blob/main/docs/TOOLS_GUIDE.md) for more details.
>To list the possible configuration parameters, add a **/config** comment.   
 

</details></td></tr>

</table>

See the [describe usage](https://github.com/Codium-ai/pr-agent/blob/main/docs/DESCRIBE.md) page for a comprehensive guide on using this tool.

</details>

### CODE REVIEW

**Comment by zafarabbas** [comment]
Task linked: [DUPLO-13189 TF Provider Docs: duplocloud_ecs_task_definition and duplocloud_ecs_service docs need to be updated](https://app.clickup.com/t/8655600/DUPLO-13189)

**Comment by zafarabbas** [comment]
Task linked: [DUPLO-6168 TF Provider: Port mappings are somehow cleared from container_definitions ECS Task definition TF resource](https://app.clickup.com/t/8655600/DUPLO-6168)

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,12 @@
+## 2024-02-09
+
+### Fixed
+- Fixed a regression in `duplocloud_k8_ingress` validation where `port` and `port_name` were not correctly validated as mutually exclusive.
```
Comment (joek-duplo) [inline CHANGELOG.md:4]: remove duplicate changelog entry

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,12 @@
...
+
+### Updated
+- Updated ECS task definition examples and documentation to use the correct `PortMappings` field instead of `ContainerMappings`.
```
Comment (joek-duplo) [inline CHANGELOG.md:7]: TF changelog entries must always have the TF resource name.  Take a look at the existing changelog for more examples.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,12 @@
...
+- Aligned the field within the port mappings from `ContainerPorts` to `ContainerPort` to match AWS supported configuration.
```
Comment (joek-duplo) [inline CHANGELOG.md:8]: TF changelog entries must always have the TF resource name.  Take a look at the existing changelog for more examples.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,12 @@
+## 2024-02-09
```
Comment (joek-duplo) [inline CHANGELOG.md:1]: duplicate date in change log.  see: https://github.com/duplocloud-internal/wiki/wiki/ProductDevBranchAndReviewModel#changelogmd-maintenance-guidelines-and-requirements

**Review by joek-duplo — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,9 @@
 ## 2024-02-09
 
+## Updated
+- Updated ECS task definition examples and documentation to use the correct `PortMappings` field instead of `ContainerMappings`.
```
Comment (joek-duplo) [inline CHANGELOG.md:4]: TF changelog entries must refer to the TF resource by name.  See: https://github.com/duplocloud-internal/wiki/wiki/ProductDevBranchAndReviewModel#changelogmd-maintenance-guidelines-and-requirements

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,9 @@
 ## 2024-02-09
 
+## Updated
+- Updated ECS task definition examples and documentation to use the correct `PortMappings` field instead of `ContainerMappings`.
+- Aligned the field within the port mappings from `ContainerPorts` to `ContainerPort` to match AWS supported configuration.
```
Comment (joek-duplo) [inline CHANGELOG.md:5]: TF changelog entries must refer to the TF resource by name.  See: https://github.com/duplocloud-internal/wiki/wiki/ProductDevBranchAndReviewModel#changelogmd-maintenance-guidelines-and-requirements

**Review by joek-duplo — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 5 above)
Comment (Matheusbafutto) [reply CHANGELOG.md]: Don't know how I missed your last comment about this. My bad

**Review by joek-duplo — APPROVED** [review]

**Pull request merged**